src/settings/ClosingReasons/ClosingReasons.js

```diff
@@ -1,46 +1,123 @@
+const fieldComponents = {
+  // eslint-disable-next-line react/prop-types
+  reason: ({ fieldProps }) => {
+    return (
+      <Field
+        {...fieldProps}
+        component={TextField}
+        fullWidth
+      />
+    );
+  },
+};
+
+const hiddenFields = ['numberOfObjects', 'lastUpdated'];
+const visibleFields = ['reason', 'source'];
+const readOnlyFields = ['source'];
+const columnMapping = {
+  reason: <FormattedMessage id="ui-orders.settings.closingReasons.reason" />,
+  source: <FormattedMessage id="ui-orders.settings.closingReasons.source" />,
+};
+
+const reasonsLabel = <FormattedMessage id="ui-orders.settings.closingReasons.reason" />;
+const formatter = {
+  // eslint-disable-next-line react/prop-types
+  reason: ({ reason }) => (
+    <FormattedMessage id={`ui-orders.settings.closingReasons.reason.${reason}`} defaultMessage={reason} />
+  ),
+  // eslint-disable-next-line react/prop-types
+  source: ({ source }) => (
+    <FormattedMessage id={`ui-orders.settings.closingReasons.source.${source}`} defaultMessage=" " />
+  ),
+};
+
+const suppressEdit = ({ source }) => source === CLOSING_REASONS_SOURCE.system;
+const suppressDelete = ({ source }) => source === CLOSING_REASONS_SOURCE.system;
+
+const ClosingReasons = ({
+  intl,
+  mutator,
+  resources,
+  stripes,
+}) => {
   return (
-    <Pane
-      defaultWidth="100%"
-      fluidContentWidth
-      paneTitle={title}
-    >
-      <AddClosingReason saveReason={saveReason} />
-      <div data-test-order-settings-closing-orders-list>
-        {
-          reasons.map(reason => (
-            <ClosingReasonItem
-              key={reason.id || reason.code}
-              id={reason.id}
-              value={reason.value}
-              isSystem={!reason.id}
-              saveReason={saveReason}
-              removeReason={removeReason}
-            />
-          ))
-        }
-      </div>
-    </Pane>
+    <ControlledVocab
+      actionSuppressor={{ edit: suppressEdit, delete: suppressDelete }}
+      baseUrl={REASONS_FOR_CLOSURE_API}
+      columnMapping={columnMapping}
+      fieldComponents={fieldComponents}
+      formatter={formatter}
+      hiddenFields={hiddenFields}
+      id="closingReasons"
+      label={intl.formatMessage({ id: 'ui-orders.settings.closingOrderReasons' })}
+      labelSingular={intl.formatMessage({ id: 'ui-orders.settings.closingReasons.reason' })}
+      mutator={mutator}
+      nameKey="reason"
+      objectLabel={reasonsLabel}
+      readOnlyFields={readOnlyFields}
+      records="reasonsForClosure"
+      resources={resources}
+      stripes={stripes}
+      visibleFields={visibleFields}
+    />
   );
 };
 
+ClosingReasons.manifest = Object.freeze({
+  values: {
+    ...reasonsForClosureResource,
+    PUT: {
+      path: `${REASONS_FOR_CLOSURE_API}/%{activeRecord.id}`,
+    },
+    DELETE: {
+      path: `${REASONS_FOR_CLOSURE_API}/%{activeRecord.id}`,
+    },
+  },
+  updaterIds: [],
+  activeRecord: {},
+  updaters: {
```

Comment:
too much copy-paste, it's definitely could be leverage into something common, it looks the same as https://github.com/folio-org/ui-orders/blob/master/src/settings/OrderNumberModifier/OrderNumberModifier.js#L93